### PR TITLE
people autocomplete: Fix it doesn't pop if `@` entered in new line.

### PR DESCRIPTION
**Pull request by jainkuniya**
We only consider `@` when the preceding char is either ` `, `\n`, `:`
or `#`. Because every email as `@` in it and we don't want to trigger
suggestions in such cases.

Fixes: #3289

image

### CODE REVIEW

File: src/autocomplete/__tests__/getAutocompleteFilter-test.js [hunk 1]
```diff
@@ -11,6 +11,12 @@ describe('getAutocompleteFilter', () => {
     selection = { start: 1, end: 1 };
     expect(getAutocompleteFilter('@', selection)).toEqual({ filter: '', lastWordPrefix: '@' });
 
+    selection = { start: 1, end: 1 };
```
Comment (borisyankov) [inline src/autocomplete/__tests__/getAutocompleteFilter-test.js]: These would be better in a new test case.

**Comment by armaanahluwalia** [comment]
Try entering more than 5 lines of text and then trying an @ mention. The popup is cutting off content at the top. You may want to look at that since you're fixing this anyway.

**Comment by jainkuniya** [comment]
image

@armaanahluwalia On Android or iOS? I can't reproduce. Can you please share the screenshot?

**Comment by borisyankov** [comment]
@armaanahluwalia The current implementation definitely has issues with height which we can improve but is surprisingly tricky so we delayed it for later.

**Comment by armaanahluwalia** [comment]
Its a pretty bad bug for a pretty common use case aka typing longer messages on zulip with an @ mention.

See [Bug Video 2](https://streamable.com/vo6ym)

**Comment by borisyankov** [comment]
Thanks for the video @armaanahluwalia
I agree it is worse than I thought and worth fixing.

File: src/autocomplete/__tests__/getAutocompleteFilter-test.js (hunk 1 above)
Comment (jainkuniya) [reply src/autocomplete/__tests__/getAutocompleteFilter-test.js]: Done 👍

**Comment by gnprice** [comment]
Thanks @jainkuniya for this fix! Merged.

This function in general is pretty convoluted and hard to follow, and this change unfortunately makes it a little bit more so. I think there are more issues lurking that are hard to see because of the complexity of it. So I'd love to see a followup to make it simpler and clearer. Filed #3345 with more details on that.

**Comment by gnprice** [comment]
@armaanahluwalia Thanks for that helpful video!

Please go ahead and file that as an issue. Here on a loosely related PR thread, it's liable to get forgotten. :wink:
